web/src/views/FindSpeaker.vue

```diff
@@ -96,19 +124,47 @@ export default {
     setError(err) {
       this.error = err;
     },
+    getPreviousPage() {
+      if (this.page > 1) {
+        this.page -= 1;
+        this.isMaxPage = false;
+      }
+    },
+    getNextPage() {
+      // leave empty, this function is overwritten by the getSpeakers method
+    },
     getSpeakers() {
       this.$db('People')
         .select({
           view: 'Published',
           sort: [{ field: 'name_en', direction: 'asc' }],
+          pageSize: this.pageSize,
         })
-        .firstPage((err, records) => {
-          if (err) {
-            this.error = err;
-          } else {
-            this.speakers = records;
-          }
-        });
+        .eachPage(
+          (records, next) => {
+            this.speakers.push(...records);
+            this.page += 1;
+            this.getNextPage = next;
+          },
+          (err) => {
+            if (err) {
+              this.error = err;
+            }
+
+            // if the error is null no new page exists
+            this.isMaxPage = true;
+
+            // reset the getNextPage function to just increment the page
+            // as there is no more results to fetch.
+            this.getNextPage = () => {
+              if (!this.isMaxPage) {
+                this.page += 1;
+                const lastPageEntry = this.page * this.pageSize;
+                this.isMaxPage = lastPageEntry >= this.speakers.length;
+              }
+            };
```

Comment:
How about leaving this on the [`getNextPage()` method](https://github.com/WWCodeTokyo/speak-her-db/pull/166/files/c988fd9aabdcebe05396463bdfd641f38522681b#r498937819) and trigger when the user actually clicks the next page?

Reply:
Ahh, I see now. Does Airtable not provide the total number of pages or entries on the request header or something?

Reply:
so the actual request to airtable returns everything we would need, the sdk just wraps it in a weird way so we have no way of knowing anymore :(